test/graphql_test.rb

```diff
@@ -374,4 +374,35 @@ def test_mutation_batch_subselection_execution
     assert_equal expected, result
     assert_equal ["Product/1,2", "Product/2,3"], queries
   end
+
+  class MyCustomExecutor < GraphQL::Batch::Executor
+    attr_reader :loaders_executed
+
+    def initialize
+      @loaders_executed = []
+      super
+    end
+
+    def around_promise_callbacks(loader)
+      @loaders_executed << loader
+      super
+    end
+  end
+
+  def test_custom_executor_class
+    schema = GraphQL::Schema.define do
+      query ::QueryType
+      mutation ::MutationType
+
+      use GraphQL::Batch, executor_class: MyCustomExecutor
+    end
+
+    custom_executor = MyCustomExecutor.new
+    MyCustomExecutor.stubs(:new).returns(custom_executor)
```

Comment:
I would prefer if you at least use a class variable for the loaders_executed array so that you don't need to mess with the implementation using stubbing.